File: data/EXP01/golang_no_system_var/2522/2522.output_2.go
func minimumPartition(s string, k int) int {
    partition := 0
    substringValue := 0
    
    for _, char := range s {
        digit := int(char - '0')
        substringValue += digit
        
        if substringValue > k {
            return -1
        }
        
        if substringValue == k {
            partition++
            substringValue = 0
        }
    }
    
    if substringValue != 0 {
        partition++
    }
    
    return partition
}

File: data/EXP01/python/2522/2522.output_2.py
# @lc app=leetcode id=2522 lang=python3
class Solution:
    def minimumPartition(self, s: str, k: int) -> int:
        n = len(s)
        dp = [float('inf')] * n
        dp[0] = 1
        for i in range(1, n):
            for j in range(i, -1, -1):
                if int(s[j:i+1]) <= k:
                    dp[i] = min(dp[i], dp[j-1] + 1)
        return dp[n-1] if dp[n-1] != float('inf') else -1